### PR TITLE
Add 2xx and 3xx response code as valid status codes

### PR DESCRIPTION
**Pull request by fredericbarthelet**
**Description**
Accept 2xx and 3xx status code response. This aims to handle 202 status code not as error.

**Tested scenarios**
Account creation

**Fixed issue**:
#127

### CODE REVIEW

**Comment by msilvagarcia** [comment]
Hey @fredericbarthelet !

Can you explain the reason of accepting 3xx status codes as well? Also, what happens when a status code 1xx is received?

File: src/Adyen/HttpClient/CurlClient.php [hunk 1]
```diff
@@ -202,7 +202,7 @@ public function requestPost(\Adyen\Service $service, $requestUrl, $params)
 
 		curl_close($ch);
 
-		if ($httpStatus != 200 && $result) {
+		if ($httpStatus >= 400 && $result) {
```
Comment (cyattilakiss) [inline src/Adyen/HttpClient/CurlClient.php]:
```suggestion
		$resultOKHttpStatusCodes = array(200, 201, 202, 203, 204);
		
		if (!in_array($httpStatus, $resultOKHttpStatusCodes) && $result) {
```

I would keep it "simple" and consider OK only the known response codes that can be returned from the API.

**Comment by fredericbarthelet** [comment]
@cyattilakiss PR updated as per your suggestion :)

**Comment by cyattilakiss** [comment]
Hi @fredericbarthelet ,

Thanks it will be merged :)

Best,
Attila

**Pull request merged**